task-sdk/src/airflow/sdk/definitions/param.py

```diff
@@ -218,6 +218,27 @@ def __getitem__(self, key: str) -> Any:
         param = self.__dict[key]
         return param.resolve(suppress_exception=self.suppress_exception)
 
+    def get(self, key: str, default: Any = None) -> Any:
+        """
+        Get the value for a key, returning default if the key is not found OR if the resolved value is None.
+
+        This method properly handles the case where a parameter exists but has a None value,
+        which should be treated as if the key doesn't exist when a default is provided.
+
+        :param key: The key to fetch
+        :param default: The default value to return if key is not found or resolved value is None
+        """
+        try:
+            param = self.__dict[key]
```

Comment:
Correct would be __dict__
```suggestion
            param = self.__dict__[key]
```